Backend/app/rider-platform/rider-app/Main/src/SharedLogic/MerchantConfig.hs

```diff
@@ -81,23 +82,27 @@ updateCustomerFraudCounters riderId merchantConfigs = Redis.withNonCriticalCross
 updateTotalRidesCounters :: (CacheFlow m r, MonadFlow m, EsqDBFlow m r, CacheFlow m r, EsqDBReplicaFlow m r, ClickhouseFlow m r) => Id Person.Person -> m ()
 updateTotalRidesCounters riderId = do
   totalRidesCount <- getTotalRidesCount riderId Nothing
-  let totalRidesCount' = totalRidesCount + 1
-  void $ QP.updateTotalRidesCount riderId (Just totalRidesCount')
+  whenJust totalRidesCount $ \count' -> do
+    QP.updateTotalRidesCount riderId (Just (count' + 1))
 
 updateTotalRidesInWindowCounters :: (CacheFlow m r, MonadFlow m) => Id Person.Person -> [DMC.MerchantConfig] -> m ()
 updateTotalRidesInWindowCounters riderId merchantConfigs = Redis.withNonCriticalCrossAppRedis $ do
   mapM_ (\mc -> incrementCount mc.id.getId mc.fraudRideCountWindow) merchantConfigs
   where
     incrementCount ind = SWC.incrementWindowCount (mkRideWindowCountKey ind riderId.getId)
 
-getTotalRidesCount :: (CacheFlow m r, MonadFlow m, EsqDBFlow m r, EsqDBReplicaFlow m r, ClickhouseFlow m r) => Id Person.Person -> Maybe DSR.SearchRequest -> m Int
+getTotalRidesCount :: (CacheFlow m r, MonadFlow m, EsqDBFlow m r, EsqDBReplicaFlow m r, ClickhouseFlow m r) => Id Person.Person -> Maybe DSR.SearchRequest -> m (Maybe Int)
 getTotalRidesCount riderId mSearchReq =
   case mSearchReq >>= DSR.totalRidesCount of
     Nothing -> do
-      totalRidesCount <- CHP.findTotalRidesCountByPersonId riderId
-      QP.updateTotalRidesCount riderId totalRidesCount
-      pure $ fromMaybe 0 totalRidesCount
-    Just totalCount -> pure totalCount
+      person <- QP.findById riderId >>= fromMaybeM (PersonNotFound ("Person not found with id: " <> riderId.getId))
+      case person.totalRidesCount of
+        Nothing -> do
+          totalRidesCount <- CHP.findTotalRidesCountByPersonId riderId
+          QP.updateTotalRidesCount riderId totalRidesCount
```

Comment:
in this case, you will endUp updating same value twice in DB if we hit this case in when called from -> `updateTotalRidesCounters`.